### PR TITLE
.Net: Ollama connector unit tests

### PR DESCRIPTION
**Pull request by BLaZeKiLL**
### Motivation and Context

This PR adds a total of 35 unit tests for the Ollama connector.

### Description

The tests are created with the HuggingFace unit tests as a reference
> GitHub Actions (CI) changes may be required to run the new test project in the pipeline

I also saw some redundancies in the HuggingFaceTextGeneration unit tests, for example, in the code below, `var contents = await sut.GetTextContentsAsync("fake-test");` is redundant, and I found this redundancy in another test (`GetStreamingTextContentsShouldHaveModelIdDefinedAsync()`) in the same file
```csharp
[Fact]
public async Task GetTextContentsShouldHaveModelIdDefinedAsync()
{
    //Arrange
    var sut = new HuggingFaceTextGenerationService("fake-model", endpoint: new Uri("https://fake-random-test-host/fake-path"), httpClient: this._httpClient);

    //Act
    var contents = await sut.GetTextContentsAsync("fake-test");
    this._messageHandlerStub.ResponseToReturn = new HttpResponseMessage(System.Net.HttpStatusCode.OK)
    {
        Content = new StringContent(@"
        [
            {
                ""generated_text"": ""Why the sky is blue? | Dept. of Science & Mathematics Education | University of Notre Dame\nWhen I was in high school I had a pretty simple conception of reality. I believed that if something made sense to me, then it must also be true. I believed that some problems were so fundamental that I couldn’t understand""
            }
        ]",
        Encoding.UTF8,
        "application/json")
    };

    // Act
    var textContent = await sut.GetTextContentAsync("Any prompt");

    // Assert
    Assert.NotNull(textContent.ModelId);
    Assert.Equal("fake-model", textContent.ModelId);
}

Also fixed one project restructuring issue which was missed in the previous PR, KernelBuilder and ServiceCollection extensions are moved to the root of the Ollama connector project
```

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/Connectors/Connectors.Ollama.UnitTests/Client/OllamaStreamJsonParserTest.cs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+using System;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using Microsoft.SemanticKernel.Connectors.Ollama.Core;
+using Xunit;
+
+namespace SemanticKernel.Connectors.Ollama.UnitTests;
+
+public class OllamaStreamJsonParserTest
```
Comment (Krzysztof318) [inline dotnet/src/Connectors/Connectors.Ollama.UnitTests/Client/OllamaStreamJsonParserTest.cs:12]: Instead of using sync version of stream parser, copy StreamJsonParser (which is async) and related unit tests from gemini branch

File: dotnet/src/Connectors/Connectors.Ollama.UnitTests/Connectors.Ollama.UnitTests.csproj [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    </None>
+    <None Update="OpenAI\TestData\*">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </None>
+    <None Update="TestData\text_generation_test_response.json">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </None>
+    <None Update="TestData\embeddings_test_response.json">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </None>
```
Comment (Krzysztof318) [inline dotnet/src/Connectors/Connectors.Ollama.UnitTests/Connectors.Ollama.UnitTests.csproj:60]: Simplify this to "testdata\*"

File: dotnet/src/Connectors/Connectors.Ollama.UnitTests/OllamaKernelBuilderExtensionsTests.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.Ollama;
+using Microsoft.SemanticKernel.Embeddings;
+using Microsoft.SemanticKernel.TextGeneration;
+using Xunit;
+
+namespace SemanticKernel.Connectors.Ollama.UnitTests;
+
+public class OllamaKernelBuilderExtensionsTests
```
Comment (Krzysztof318) [inline dotnet/src/Connectors/Connectors.Ollama.UnitTests/OllamaKernelBuilderExtensionsTests.cs:13]:
```suggestion
/// Unit tests for <see cref="<kernel builder extensions class>
public class OllamaKernelBuilderExtensionsTests
```

File: dotnet/src/Connectors/Connectors.Ollama.UnitTests/OllamaServiceCollectionExtensionsTests.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.Ollama;
+using Microsoft.SemanticKernel.Embeddings;
+using Microsoft.SemanticKernel.TextGeneration;
+using Xunit;
+
+namespace SemanticKernel.Connectors.Ollama.UnitTests;
+
+public class OllamaServiceCollectionExtensionsTests
```
Comment (Krzysztof318) [inline dotnet/src/Connectors/Connectors.Ollama.UnitTests/OllamaServiceCollectionExtensionsTests.cs:13]: same add docs

File: dotnet/src/Connectors/Connectors.Ollama.UnitTests/Services/OllamaChatCompletionTests.cs [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.Ollama;
+using Microsoft.SemanticKernel.Connectors.Ollama.Core;
+using Xunit;
+
+namespace SemanticKernel.Connectors.Ollama.UnitTests;
+
+public class OllamaChatCompletionTests : IDisposable
```
Comment (Krzysztof318) [inline dotnet/src/Connectors/Connectors.Ollama.UnitTests/Services/OllamaChatCompletionTests.cs:17]: Make this class more compact. Blank lines between Arrange/act/assert is sufficient. Remove blank lines between single statements.

File: dotnet/src/Connectors/Connectors.Ollama.UnitTests/Services/OllamaTextEmbeddingGenerationTests.cs [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Connectors.Ollama;
+using Microsoft.SemanticKernel.Connectors.Ollama.Core;
+using Xunit;
+
+namespace SemanticKernel.Connectors.Ollama.UnitTests;
+
+public class OllamaTextEmbeddingGenerationTests : IDisposable
+{
```
Comment (Krzysztof318) [inline dotnet/src/Connectors/Connectors.Ollama.UnitTests/Services/OllamaTextEmbeddingGenerationTests.cs:16]: same compact code; blank lines

File: dotnet/src/Connectors/Connectors.Ollama.UnitTests/Services/OllamaTextGenerationTests.cs [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Connectors.Ollama;
+using Microsoft.SemanticKernel.Connectors.Ollama.Core;
+using Microsoft.SemanticKernel.TextGeneration;
+using Xunit;
+
+namespace SemanticKernel.Connectors.Ollama.UnitTests;
+
+public class OllamaTextGenerationTests : IDisposable
+{
```
Comment (Krzysztof318) [inline dotnet/src/Connectors/Connectors.Ollama.UnitTests/Services/OllamaTextGenerationTests.cs:18]: compact...

File: dotnet/src/Connectors/Connectors.Ollama/Client/OllamaChatRequest.cs [hunk 8]
```diff
@@ -10,7 +10,7 @@ namespace Microsoft.SemanticKernel.Connectors.Ollama.Core;
 /// <summary>
 /// Ollama chat request object.
 /// </summary>
-internal sealed class OllamaChatRequest
+public sealed class OllamaChatRequest
 {
```
Comment (Krzysztof318) [inline dotnet/src/Connectors/Connectors.Ollama/Client/OllamaChatRequest.cs:14]: Why did you change this class to public? If only for unit tests, use <internalsvisibleto> instead of making class public.

File: dotnet/src/Connectors/Connectors.Ollama/Client/OllamaChatRequest.cs [hunk 9]
```diff
@@ -51,7 +51,7 @@ internal static OllamaChatRequest FromPromptAndExecutionSettings(ChatHistory cha
         };
     }
 
-    internal sealed class OllamaChatRequestMessage
+    public sealed class OllamaChatRequestMessage
```
Comment (Krzysztof318) [inline dotnet/src/Connectors/Connectors.Ollama/Client/OllamaChatRequest.cs:54]: same why public?

File: dotnet/src/Connectors/Connectors.Ollama/OllamaPromptExecutionSettings.cs [hunk 10]
```diff
@@ -46,5 +47,6 @@ public static OllamaPromptExecutionSettings FromExecutionSettings(PromptExecutio
     /// <summary>
     /// Maximum number of tokens to generate.
     /// </summary>
+    [JsonPropertyName("max_tokens")]
     public int? MaxTokens { get; set; }
```
Comment (Krzysztof318) [inline dotnet/src/Connectors/Connectors.Ollama/OllamaPromptExecutionSettings.cs:51]: adjust prompt execution settings to new pattern used actual in openai connector.

**Review by Krzysztof318 — CHANGES_REQUESTED** [review]

File: dotnet/src/Connectors/Connectors.Ollama.UnitTests/Client/OllamaStreamJsonParserTest.cs (hunk 1 above)
Comment (RogerBarreto) [reply dotnet/src/Connectors/Connectors.Ollama.UnitTests/Client/OllamaStreamJsonParserTest.cs:12]: This can be in a next PR. This is UT focused for now.

@BLaZeKiLL just drop the JsonParser tests as they will be done by the SemanticKernel.UnitTests.

**Comment by BLaZeKiLL** [comment]
@Krzysztof318 Thanks for the review, I'll keep these points in mind on future PR's
